### PR TITLE
Use common sessioner for API and web routes

### LINKED ISSUE #18070: Memory session provider does not return private repositories in dashboard search
### Gitea Version

1.15.8

### Git Version

git version 2.30.1 (Apple Git-130)

### Operating System

macOS 11.6

### How are you running Gitea?

Running gitea with a pre-built download from the releases page (https://github.com/go-gitea/gitea/releases/tag/v1.15.8).

Running locally, ran the installer and selected a SQLite database and "local mode"

After installation, changed the session provider from "file" to "memory"

### Database

SQLite

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

https://gist.github.com/jprjr/36ffebe2652bfc68ac4e24c67e2841e1

### Description

When using the "memory" provider, the dashboard page's repository list (and search results) does not include private repositories created by the logged-in user.

Viewing the profile page does show them, it only seems to affect the search box on the dashboard page.

### Screenshots

Here is a screenshot with the session provider on the default, "file":

image

And here is a screenshot with the session provider changed to "memory" - notice the private repo is not listed.

image

### PR DESCRIPTION
**Pull request by zeripath**
Since the regenerate session ID PR in 1.15.8 - some users of the memory session provider have been
reporting difficulties with getting API results.

I am uncertain as to why this is happening - but I think that the sessioner being
created twice may be a potential cause for this. Therefore this PR attempts to move
this out to a common sessioner as it is in 1.16.

Fix #18070

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by zeripath** [comment]
Please note I cannot reproduce this bug myself directly even with memory session provider - therefore those who are suffering this problem SHOULD check that this solves the issue.

File: routers/init.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ import (
 	pull_service "code.gitea.io/gitea/services/pull"
 	"code.gitea.io/gitea/services/repository"
 	"code.gitea.io/gitea/services/webhook"
+	"gitea.com/go-chi/session"
```
Comment (lunny) [inline routers/init.go:44]: blank line.

**Review by lunny — APPROVED** [review]

File: routers/init.go (hunk 1 above)
Comment (zeripath) [reply routers/init.go:44]:
```suggestion

	"gitea.com/go-chi/session"
```

**Comment by jprjr** [comment]
@zeripath confirming - this resolves the issue for me

**Comment by perfectra1n** [comment]
I was an affected user, having to change the session provider to file. The fix above worked for me, and repos can be seen from the dashboard.

**Review by KN4CK3R — APPROVED** [review]

**Pull request merged**